rpcs3/rpcs3qt/trophy_manager_dialog.cpp

```diff
@@ -581,18 +585,26 @@ void trophy_manager_dialog::ShowContextMenu(const QPoint& loc)
 	menu->exec(globalPos);
 }
 
-void trophy_manager_dialog::PopulateTrophyDB()
+void trophy_manager_dialog::PopulateTrophyDB(QProgressDialog *progressDialog)
 {
 	m_trophies_db.clear();
 
-	QDirIterator dir_iter(qstr(vfs::get(m_trophy_dir)), QDir::Dirs | QDir::NoDotAndDotDot);
-	while (dir_iter.hasNext())
+	QDir trophyDir(qstr(vfs::get(m_trophy_dir)));
+	const auto folderList = trophyDir.entryList(QDir::Dirs | QDir::NoDotAndDotDot);
+	const int count = folderList.count();
+	progressDialog->setMaximum(count);
+
+	for (int i = 0 ; i < count; i++)
 	{
-		dir_iter.next();
-		std::string dirName = sstr(dir_iter.fileName());
+		std::string dirName = sstr(folderList.value(i));
 		LOG_TRACE(GENERAL, "Loading trophy dir: %s", dirName);
 		LoadTrophyFolderToDB(dirName);
+		progressDialog->setValue(i);
 	}
+
+	RepaintUI();
+	progressDialog->close();
+	delete progressDialog;
```

Comment:
this doesn't actually fully remove the dialog.
use `progressDialog->deleteLater();` instead